contracts/protocol/libraries/logic/GenericLogic.sol

```diff
@@ -152,14 +157,15 @@ library GenericLogic {
     DataTypes.UserConfigurationMap memory userConfig,
     mapping(uint256 => address) storage reserves,
     uint256 reservesCount,
-    address oracle
+    Feeds memory feeds
   ) internal view returns (uint256, uint256, uint256, uint256, uint256) {
     CalculateUserAccountDataVars memory vars;
 
     if (userConfig.isEmpty()) {
       return (0, 0, 0, 0, uint256(-1));
     }
-    for (vars.i = 0; vars.i < reservesCount; vars.i++) {
+
+    for (vars.i = reservesCount - 1; vars.i < reservesCount; vars.i--) {
```

Comment:
at vars.i = 0 it skips the last reserve
and vars.i < reservesCount will always be true

Reply:
vars.i = 0 is reserved for ithaca collateral, so we are not considering it to be used for any other aave collateral

Reply:
reason for `vars.i < reservesCount` is that it fails the `vars.i = -1` check and is only true for real reserve values.

Reply:
how to we reserve vars.i = 0 for ithaca collateral?

Reply:
added a test in https://github.com/ithaca-protocol/protocol-v2/pull/3/commits/a2a4fbf637e7a631637a49bc5248cbfe1a7f1506